wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java

```diff
@@ -260,34 +263,67 @@ public NodeList getFileData(String logicalname, Long beginrow, Integer numrows,
     }
 
     /**
-     * Use this function to retrieve file metadata such as column information
+     * Use this function to retrieve file metadata such as column information,
+     * for superfiles the metadata from the first subfile will be returned.
      * @param logicalname    - Logical filename.
      * @param clustername    - Optional - The cluster the logical filename is associated with.
-     * @return Array of DFUDataColumns
+     * @return ArrayList of DFUDataColumnInfo
      * @throws Exception
      */
-    public DFUDataColumn[] getFileMetaData(String logicalname, String clustername) throws Exception
+    public List<DFUDataColumnInfo> getFileMetaData(String logicalname, String clustername) throws Exception
     {
         WsDfuServiceSoapProxy proxy = getSoapProxy();
-        DFUGetFileMetaDataRequest req = new DFUGetFileMetaDataRequest();
-        DFUDataColumn[] cols = null;
-
-        req.setLogicalFileName(logicalname);
-
-        if (clustername != null)
-            req.setClusterName(clustername);
 
+        List<DFUDataColumnInfo> cols=new ArrayList<DFUDataColumnInfo>();
+        String eclrecord=null;
+        //getFileMetadata fails for superfiles; use first subfile to retrieve record structure if this is the case
```

Comment:
There might be legacy superfiles w/ multiple subfiles with different rec defs, we might want to interrogate all subfiles and confirm they are homogeneous. If not, throw exception?